internal/serialization/serialization.go

```diff
@@ -109,18 +116,15 @@ func (s *Service) WriteObject(output pubserialization.DataOutput, object interfa
 
 func (s *Service) ReadObject(input pubserialization.DataInput) interface{} {
 	serializerID := input.ReadInt32()
-	serializer := s.registry[serializerID]
+	serializer := s.lookupBuiltinDeserializer(serializerID)
+	if serializer == nil {
+		serializer = s.registry[serializerID]
```

Comment:
Can this be `nil`?

Reply:
Yes, it can. A panic is thrown but is caught and converted to an error before it reaches user code. I've updated that to explicitly panic with a more descriptive message at: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/709d98b5690a4f8fbe20e86cec31017edf566068